src/components/link/link.component.tsx

```diff
@@ -84,6 +84,12 @@ export const Link = React.forwardRef<
     const [hasFocus, setHasFocus] = useState(false);
     const l = useLocale();
     const { inMenu } = useContext(MenuContext);
+    const internalRef = useRef<HTMLLinkElement | HTMLButtonElement>(null);
```

Comment:
**suggestion**: the bug seems to surface when the focused `Link` is either disabled or has its hyperlink removed immediately after and the state is not updated.

What do you think about using an effect to handle this? This way we wouldn't need to use refs:

```tsx
useEffect(() => {
      if ((disabled || !href) && hasFocus) setHasFocus(false);
}, [disabled, hasFocus, href]);
```

We could avoid an extra re-render by adding logic to `Pager`, `Breadcrumbs`, etc. But I expect it could become unnecessarily complex to handle all possible scenarios where focused `Link` could be disabled immediately after.

Reply:
yep good point, I'd missed that `isCurrent` negated the href and onClick. I'm not keen on adding them in the parent components as we'd need to handle it every time we add a new one that composes Link etc so fixing it as low down as possible is the better approach.

I think the following will suffice as we don't really care about the current state and if we do we can handle it in a callback within the state setter

```
useEffect(() => {
      if (disabled || !(href && onClick)) {
        setHasFocus(false);
      }
    }, [disabled, href, onClick]);